.github/workflows/actionlint.yml

```diff
@@ -7,29 +7,46 @@ on:
       - "main"
       - "release-**"
     paths:
-      - '.github/**'
+      - '.github/workflows/*.ya?ml'
+      - '.github/workflows/actionlint.*' # This workflow
   pull_request:
     branches:
       - "main"
       - "release-**"
     paths:
-      - '.github/**'
+      - '.github/workflows/*.ya?ml'
+      - '.github/workflows/actionlint.*' # This workflow
+
+env:
+  LC_ALL: en_US.UTF-8
 
 defaults:
   run:
     shell: bash
 
+permissions:
+  contents: read
+
 jobs:
   actionlint:
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout
-        uses: actions/checkout@v4
+      - name: "Harden Runner"
+        uses: step-security/harden-runner@f086349bfa2bd1361f7909c78558e816508cdc10 # v2.8.0
+        with:
+          egress-policy: audit # TODO: change to 'egress-policy: block' after couple of runs
```

Comment:
@bjhargrave do you think we could change this since this has run many times?

Reply:
As above, I would like to merge using audit and then follow up with replacing audit in a later PR.